src/Agent.Worker/JobExtension.cs

```diff
@@ -131,17 +131,27 @@ public async Task<JobInitializeResult> InitializeJob(IExecutionContext jobContex
                     Stack<IStep> postJobStepsBuilder = new Stack<IStep>();
                     Dictionary<Guid, Variables> taskVariablesMapping = new Dictionary<Guid, Variables>();
 
-                    if (context.Container != null)
+                    if (context.Container != null || context.SidecarContainers?.Count > 0)
```

Comment:
make sure context.SidecarContainers is never null, so we don't have to `?.` all the time.

Reply:
Cleaned up